packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -78,14 +78,14 @@ export class TreeActions {
             }),
             switchMap((props) =>
               this._loader.loadNodes(props.loadOptions).pipe(
-                collectTreePartsUntil(this._reset, props.initialRootNode),
+                collectTreePartsUntil(this._reset, props.loadOptions.parent, props.initialRootNode),
                 tap({
                   next: (newModel: TreeModel) => {
                     const childNodes = newModel.parentChildMap.get(props.parentId);
                     const firstChildNode = childNodes?.length ? newModel.idToNode.get(childNodes[0]) : undefined;
                     this.handleLoadedHierarchy(props.parentId, newModel);
                     // only report load duration if no error occurs
-                    if (!(firstChildNode && isTreeModelInfoNode(firstChildNode))) {
+                    if (!firstChildNode) {
```

Comment:
doesn't look like the comment above makes sense anymore

Reply:
This part overall seems to be wrong, changed to reflect previous implementation.